### PR TITLE
Use constants and minor fixes by luke-jr

### PR DESCRIPTION
**Pull request by maflcko**
Not sure if #6349 finds its way out of rebase hell anytime soon. Those three commits can be merged without rebase (I preserved the commit hashes)

They already got some sort of code review as part of #6349 so it's better to merge them without rebase, imo.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Needs a minor fix, thus rebase. Closing.

File: src/rpcmining.cpp [hunk 1]
```diff
@@ -211,7 +211,7 @@ UniValue setgenerate(const UniValue& params, bool fHelp)
     if (params.size() > 0)
         fGenerate = params[0].get_bool();
 
-    int nGenProcLimit = -1;
+    int nGenProcLimit = GetArg("-genproclimit", DEFAULT_GENERATE_THREADS);
```
Comment (jtimon) [inline src/rpcmining.cpp]: Why this changes from -1 to ```GetArg("-genproclimit", DEFAULT_GENERATE_THREADS)``` but in ```getmininginfo()``` the change is from -1 to DEFAULT_GENERATE_THREADS?

Comment (MarcoFalke) [inline src/rpcmining.cpp]: You still want to obey the command line arg if nothing is set via rpc.

File: src/rpcblockchain.cpp [hunk 2]
```diff
@@ -539,13 +539,15 @@ UniValue gettxout(const UniValue& params, bool fHelp)
...
+    int nCheckLevel = GetArg("-checklevel", DEFAULT_CHECKLEVEL);
+    int nCheckDepth = GetArg("-checkblocks", DEFAULT_CHECKBLOCKS);
     if (fHelp || params.size() > 2)
         throw runtime_error(
             "verifychain ( checklevel numblocks )\n"
             "\nVerifies blockchain database.\n"
             "\nArguments:\n"
-            "1. checklevel   (numeric, optional, 0-4, default=3) How thorough the block verification is.\n"
-            "2. numblocks    (numeric, optional, default=288, 0=all) The number of blocks to check.\n"
+            "1. checklevel   (numeric, optional, 0-4, default=" + strprintf("%d", nCheckLevel) + ") How thorough the block verification is.\n"
```
Comment (jonasschnelli) [inline src/rpcblockchain.cpp]: hmm... is "default" the correct context here? Because it's no longer the default value if you have provided a different value over the command line argument. Maybe `current default value`?

**Comment by jonasschnelli** [comment]
I think this is a good idea.
utACK.

Miner and Wallet init help strings could also slowly migrate into wallet/ miner/ classes. I once did this for the wallet, but it requires rebase and maybe more small, reviewable PRs: https://github.com/bitcoin/bitcoin/pull/5990

**Comment by dcousens** [comment]
utACK

File: src/rpcmining.cpp (hunk 1 above)
Comment (jtimon) [inline src/rpcmining.cpp]: Is it a bugfix or enhanced functionality? Again, why one uses GetArg() but the other doesn't?

File: src/rpcblockchain.cpp (hunk 2 above)
Comment (jtimon) [inline src/rpcblockchain.cpp]: The default value means "the default value unless you set it via command line argument".

**Comment by jtimon** [comment]
utACK besides my nit/question.

File: src/rpcmining.cpp (hunk 1 above)
Comment (MarcoFalke) [inline src/rpcmining.cpp]: Yes, it fixes one bug:  If genproclimit is omitted to RPC setgenerate, don't change it; (copied from commit msg)

Comment (MarcoFalke) [inline src/rpcmining.cpp]: Making `getmininginfo` push back the "correct" value `nGenProcLimit` may involve some more work. But at least this PR did not introduce the issue. The PR is not perfect, but improves things considerably.

Comment (jtimon) [inline src/rpcmining.cpp]: Thank you. Everything else was completely trivial, but I thought this deserved a more detailed explanation.

File: src/rpcblockchain.cpp (hunk 2 above)
Comment (luke-jr) [inline src/rpcblockchain.cpp]: "default" means whatever you get if you don't provide it.

File: src/init.cpp [hunk 3]
```diff
@@ -404,12 +404,16 @@ std::string HelpMessage(HelpMessageMode mode)
     if (showDebug)
     {
         strUsage += HelpMessageOpt("-checkpoints", strprintf("Disable expensive verification for known chain history (default: %u)", 1));
-        strUsage += HelpMessageOpt("-dblogsize=<n>", strprintf("Flush database activity from memory pool to disk log every <n> megabytes (default: %u)", 100));
+#ifdef ENABLE_WALLET
+        strUsage += HelpMessageOpt("-dblogsize=<n>", strprintf("Flush wallet database activity from memory to disk log every <n> megabytes (default: %u)", DEFAULT_WALLET_DBLOGSIZE));
```
Comment (laanwj) [inline src/init.cpp]: Good find on missing ENABLE_WALLET here

**Comment by laanwj** [comment]
utACK

File: src/init.cpp (hunk 3 above)
Comment (MarcoFalke) [inline src/init.cpp]: All credit goes to @luke-jr ;)
